core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -289,7 +294,7 @@ object TopicCommand extends Logging {
         if (topic.hasReplicaAssignment)
           adminZkClient.createTopicWithAssignment(topic.name, topic.configsToAdd, topic.replicaAssignment.get)
         else
-          adminZkClient.createTopic(topic.name, topic.partitions.get, topic.replicationFactor, topic.configsToAdd, topic.rackAwareMode)
+          adminZkClient.createTopic(topic.name, topic.partitions.get, topic.replicationFactor.get, topic.configsToAdd, topic.rackAwareMode)
```

Comment:
The error may be a bit obscure if no `--partitions` or `--replication-factor` option was provided. Could we detect this case in the argument checks we have below?

Reply:
I wasn't sure if it was worth adding bloat for the deprecated code, but I'm happy adding it back (I had it then removed it)